modules/ballerina-lang/src/main/java/org/ballerinalang/repository/HierarchicalPackageRepository.java

```diff
@@ -93,9 +93,17 @@ private boolean isSystemPackage(PackageID pkgID) {
     @Override
     public Set<PackageID> listPackages(int maxDepth) {
         if (this.cachedPackageIds == null) {
-            this.cachedPackageIds = new LinkedHashSet<>(this.systemRepo.listPackages(maxDepth));
-            if (this.parentRepo != null) {
-                this.cachedPackageIds.addAll(this.parentRepo.listPackages(maxDepth));
+            if (this.systemRepo == null) {
```

Comment:
Check earlier parentRepo == null possibility. If even that's the case, no need to have too many if/else conditions as such, simply create an empty LinkedHashSet and two if conditions to check if repo variable are not null, and if so, just do addAll operations on them.

Reply:
I did it that way because it has slightly better performance due to initialCapacity. +1 changing it to look better.